### PR TITLE
Explore: Use Datasource Onboarding page when visiting without any datasource set up

### LINKED ISSUE #57652: Redesign Explore empty page
Starting with #58218, we are taking some initiatives to help users onboarding when setting up Grafana for the first time.

Similarly to dashboards, when a user navigates to Explore without having any data source set up we want to redirect them to the datasources page in order to guide them towards setting up their first data source.

In #58745, an "empty state" has been proposed for dashboards ( [Figma Design](https://www.figma.com/file/RqCjSrm8uj0Xa003mfAfel/Dashboards-G10?node-id=35%3A936&t=R8ngm3OTZ2jWmBPk-0) ). After a brief discussion with @thanos-karachalios we figured that Explore can use the same design with minor adjustments (copy must be changed to reference Explore instead of Dashboards, additionally what currently reads "Or set up a new dashboard with sample data" should instead point to Explore and allow users to run queries on sample data)  and contribute to Dashboard's goals.

#58795 introduces a `datasourceOnboarding` feature flag that enables said empty state page when navigating to dashboards. We can use the same for Explore.

(maybe) blocked by #55450

### PR DESCRIPTION
**Pull request by Elfo404**
**What is this feature?**

Like https://github.com/grafana/grafana/pull/58795, adds an onboarding page to Explore when a user navigates to it without having any datasource set up.

Fixes #57652

**Special notes for your reviewer**:

i renamed `Wrapper.tsx` (and test file) to `ExplorePage` to better describe it (and to avoid to create a "wrapper" of a "wrapper").
changes in `public/app/features/explore/ExplorePage.test.tsx` are unrelated, but they solve better issue that popped up by renaming the file.

for @grafana/dashboards-squad:
i changed the Datasource Onboarding component to be slightly more generic. i'm not super sure about the naming of `CTAText` and `onCTAClick` props, an alternative may be `dismissText` and `onDismiss` which may better reflect what that button is.
Also something kinda meh is importing it directly from dashboard, do you have any better idea on how we can share that component?

**How to test this:**
enable `datasourceOnboarding` feature flag and make sure there are no datasource set up in Grafana.

### CODE REVIEW

File: public/app/features/dashboard/containers/NewDashboardPage.tsx [hunk 1]
```diff
@@ -25,6 +26,13 @@ export default function NewDashboardPage(props: GrafanaRouteComponentProps) {
   return showDashboardPage ? (
     <DashboardPage {...props} />
   ) : (
-    <DatasourceOnboarding onNewDashboard={() => setCreateDashboard(true)} loading={loading} />
+    <DatasourceOnboarding
```
Comment (ivanortegaalba) [inline public/app/features/dashboard/containers/NewDashboardPage.tsx:29]: Super small naming suggestion. WDYT?
```suggestion
    <EmptyStateNoDatasource
```

**Review by ivanortegaalba — APPROVED** [review]
From the Dashboards side, everything looks good so far 😄 Thanks Gio!

**Comment by ivanortegaalba** [comment]
Hi @Elfo404, 

> Also something kinda meh is importing it directly from dashboard, do you have any better idea on how we can share that component?

What about moving it into `/features/datasource/` and both projects import from there? I think the component fits better there.

**Comment by L-M-K-B** [comment]
It might be a dumb question but I would like to know how to make this new page visible in my local dev environment. I looked it up in feature flags but couldn't find it. If it's there which name does it have?

**Comment by Elfo404** [comment]
@L-M-K-B the feature flag is `datasourceOnboarding`

**Review by gelicia — APPROVED** [review]
Tested with the feature flag on and off. 

Was just wondering though, why a feature flag? If this is providing more functionality than what we have now, couldn't we just turn it on?

File: public/app/features/explore/EmptyStateWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import { config } from '@grafana/runtime';
+import { GrafanaRouteComponentProps } from 'app/core/navigation/types';
+import { EmptyStateNoDatasource } from 'app/features/datasources/components/EmptyStateNoDatasource';
+import { ExploreQueryParams, useDispatch, useSelector } from 'app/types';
+
+import { loadDataSources } from '../datasources/state';
+
+import { ExplorePage } from './ExplorePage';
+
+export default function NewDashboardPage(props: GrafanaRouteComponentProps<{}, ExploreQueryParams>) {
```
[the remark this answers is not in the document]
Comment (Elfo404) [reply public/app/features/explore/EmptyStateWrapper.tsx:13]: good point, will change

**Comment by Elfo404** [comment]
@gelicia mostly because it's consistent with what dashboard is doing atm. i think we can also remove the flag from explore if we want but i'd personally leave it to roll it put all at once. wdyt?

**Comment by gelicia** [comment]
> @gelicia mostly because it's consistent with what dashboard is doing atm. i think we can also remove the flag from explore if we want but i'd personally leave it to roll it put all at once. wdyt?

Oh, no, that makes perfect sense. Thank you!

**Pull request merged**